pages/data-design/branches/pulling-pushing-items-between-branches.mdx

```diff
@@ -0,0 +1,113 @@
+
+![Navigating to tracking plan changes](/images/pulling-pushing-items/navigating-to-tracking-plan-changes.png)
+
+There is an button in the top right corner of every tracking plan change that can be pushed to another branch.
+
+![Pressing push button](/images/pulling-pushing-items/press-push-button.png)
+
+When that button is pushed, you will get a modal that prompts you to pick a branch to push the changes to and review your selection.
+You have to first pick a branch so that Avo can accurately display how the changes will be applied, as it might differ per branch if the item already exists on the destination branch.
+
+<center><img
+  src="/docs/images/pulling-pushing-items/empty-push-modal.png"
+  width="60%"
+  alt="Empty review modal"
+/></center>
+
+When the destination branch is selected, the actual diff for the destination branch is shown. The items shown can be expanded to show the details of what will be added to the destination branch.
+If the item you selected to push is an event or a property bundle, any related property changes are displayed below. See the [property changes related to event](#property-changes-related-to-event) section for details.
+Once you have reviewed your selection, you can press the Push to branch button and the changes will be applied to the destination branch.
+
+<center><img
+  src="/docs/images/pulling-pushing-items/collapsed-diff-modal.png"
+  width="60%"
+  alt="Empty review modal"
+/></center>
+
+Then you can navigate to your destination branch and use the item you pushed there.
+
+![Empty review modal](/images/pulling-pushing-items/destination-branch.png)
+
+## Pulling items from another branch (coming in March 2025)
+
+Soon it will be possible to search and get results from other branches. Those item can be pulled to the branch the user is on, and can be used on that branch.
+
+## Merging branches where items have been pushed or pulled
+
+When either the source or destination branch where items have been pushed or pulled are merged to main, you will be prompted to pull main on the branch where the items still remain.
+As the ID of the items are the same, the items exist on main and there is no need to keep them specifically on the other branch. 
+
+
+## Property changes related to event
+
+When an event or property bundle is selected to push to another branch, the review modal will show a list of properties related to the event or property bundle.
+By default, all related properties and their changes are kept on the event or property bundle.
+
+For events, it's possible to toggle whether the related properties and their changes are pushed too. The related property changes are the following:
+- Any **new properties** that were added to the selected item and **do not exist** on the destination branch
+    - When toggled off on events, the property will be excluded on the destination branch
```

Comment:
Not entirely true, the property will be on the event but it won't exist on the other branch 🤷‍♂️

Reply:
Ok so the relationship is kept behind the scenes? So if the destination branch is merged, then it will still stay on the source branch when main is pulled? Or what? Can you help me reframe it to describe what the user experiences?

Reply:
Yes exactly, in any case (even though we would omit adding the not existing property) it would reconcile to include it when everything has been merged. I don't think we need to say anything more here, I just wanted to mention it in case it shed some more light on how this works in the background.